### PR TITLE
Drop Python 2.7 support and remove 2.7 syntax

### PR DESCRIPTION
**Pull request by jace**

### CODE REVIEW

**Comment by jace** [comment]
Some additional changes in either 0.7.0 or 0.7.1:

1. Remove the obsolete docflow module. No current project uses it
2. Remove the make_password and check_password functions. passlib is better
3. Convert tests to pytest `assert` syntax
4. Move doctests to unit tests
5. Use f-strings for formatting as they are standard in Py 3.6+
6. Add type hints

**Comment by jace** [comment]
Deprecations should happen in major rather than minor releases, so I'll add that to 0.7.0.

File: coaster/utils/text.py [hunk 1]
```diff
@@ -372,20 +359,9 @@ def simplify_text(text):
...
-            text = text.translate(text.maketrans("", "", string.punctuation)).lower()
-        else:  # pragma: no cover
-            text = six.text_type(
-                text.encode('utf-8')
-                .translate(string.maketrans("", ""), string.punctuation)
-                .lower(),
-                'utf-8',
-            )
-    else:
-        text = text.translate(string.maketrans("", ""), string.punctuation).lower()
```
Comment (iambibhas) [inline coaster/utils/text.py:390]: This case is no longer valid?

Comment (jace) [reply coaster/utils/text.py:390]: `bytes` is no longer a valid arg type.

**Pull request merged**